### ISSUE TITLE
Request: add toggle button on window menu "stay on top"

### ISSUE THREAD

**Issue by AndroidDeveloperLB** [Genymobile/scrcpy#60]
As the title says.

**Comment by biggestdickus** [comment]
Re: Add toggle button on window menu "stay on top"
I second the motion. I use it to watch TV on my Computer at the corner of monitor (like Picture-in Picture). Needs to stay on top when working on computer.

**Comment by npes87184** [comment]
There is a flag `SDL_WINDOW_ALWAYS_ON_TOP` can be set. However, looks like it cannot be toggle after Window created.

**Comment by AndroidDeveloperLB** [comment]
Yes, but it's quite annoying to have all sorts of parameters instead of a decent UI that will remember its settings.
I know this tool is mainly for developers, but even for developers, Google provides a nice UI for its emulator, including even stay-on-top setting.

**Comment by rom1v** [comment]
> it's quite annoying to have all sorts of parameters instead of a decent UI

A decent UI with real widgets could be useful for adding boutons and other things, but concretely this means replacing C/SDL by C++/Qt to get a multiplatform app.

This was the first option I considered when I started scrcpy (in part because Qt is the framework I know the most), but this would be less lightweight, and one reason I did't choose Qt is that using Qt+Qml would require an additional unnecessary thread for rendering the video, adding synchronization complexity (and maybe latency). It would still be possible to use the QtWidgets API though (like in VLC).

The lightweightness was one goal of scrcpy, and I wanted no border around the device anyway, so I chose SDL.

Then many people come with their own needs, but there are sometimes trade-offs to make. Also, replacing the UI framework would be mostly a rewrite of the client app.

By the way, IMO "always on top" should be an option of the windows manager, which can be applied to every window without any code required from applications. This is what XFCE provides, also probably KDE, and Gnome used to provide this option. Unfortunately, some others don't (like that of Windows).

**Comment by biggestdickus** [comment]
I already have AutoHotKey install installed on Windows. Found this one line script for Allows-on-Top:
https://www.howtogeek.com/196958/the-3-best-ways-to-make-a-window-always-on-top-on-windows/

**Comment by AndroidDeveloperLB** [comment]
@rom1v Supposed you had a context menu on the window title.
Would this be hard and still require extra libraries?

What if you had an extra app for the settings? Even in console?
